addons/edi/models/ir_attachment.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+
+import fnmatch
+
+class IrAttachment(models.Model):
+
+    _inherit = 'ir.attachment'
+
+    @api.constrains('res_model')
+    def check_filename(self):
+        EdiDocument = self.env["edi.document"]
+        EdiGatewayPath = self.env["edi.gateway.path"]
+        self.ensure_one()
+        if self.res_model == 'edi.document':
+            doc = EdiDocument.browse(self.res_id)
+            if doc.doc_type_id.enforce_filename:
+                recs = EdiGatewayPath.search([('doc_type_ids', '=', doc.doc_type_id.id)])
```

Comment:
If document type doesn't have an EDI Gateway or path, (i.e. if `recs` is empty). This code will always raise a `ValidationError`. I think we should bypass the rest of the code if `recs == False`